File: project/app/src/main/java/org/owntracks/android/ui/welcome/permission/PermissionFragmentViewModel.java
package org.owntracks.android.ui.welcome.permission;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import org.owntracks.android.injection.scopes.PerActivity;
import org.owntracks.android.ui.base.viewmodel.BaseViewModel;

import javax.inject.Inject;


@PerActivity
public class PermissionFragmentViewModel extends BaseViewModel<PermissionFragmentMvvm.View> implements PermissionFragmentMvvm.ViewModel<PermissionFragmentMvvm.View> {
    private boolean permissionGranted;

    @Inject
    PermissionFragmentViewModel() {

    }

    @Override
    public void attachView(@NonNull PermissionFragmentMvvm.View view, @Nullable Bundle savedInstanceState) {
        super.attachView(view, savedInstanceState);
    }

    @Override
    public void onFixClicked() {
        getView().requestFix();
    }

    public boolean isPermissionGranted() {
        return permissionGranted;
    }

    public void setPermissionGranted(boolean permissionGranted) {
        this.permissionGranted = permissionGranted;
        notifyChange();
    }
}

File: project/app/src/main/java/org/owntracks/android/ui/welcome/permission/PermissionFragmentViewModel.kt
package org.owntracks.android.ui.welcome.permission

import androidx.databinding.Bindable
import dagger.hilt.android.scopes.ActivityScoped
import org.owntracks.android.BR
import org.owntracks.android.ui.base.viewmodel.BaseViewModel
import javax.inject.Inject

@ActivityScoped
class PermissionFragmentViewModel @Inject internal constructor() :
    BaseViewModel<PermissionFragmentMvvm.View?>() {
    @get:Bindable
    var isPermissionGranted = false
        set(permissionGranted) {
            field = permissionGranted
            notifyPropertyChanged(BR.permissionGranted)
        }

    @get:Bindable
    var isPermissionRequired = false
        set(permissionRequired) {
            field = permissionRequired
            notifyPropertyChanged(BR.permissionRequired)
        }

    fun onFixClicked() {
        view!!.requestFix()
    }
}

